server.js

```diff
@@ -1202,9 +1202,10 @@ app.get('/get-task/:version(\\d+)', asyncMiddleware( async (req, res, next) => {
 
         // Not the most robust, but if we have completed and send out more than X requests for this match, stop requesting it.
         //
-        pending_matches[pending_matches.length - 1].game_count++;
+        match.game_count++;
+        match.requests.push(now);
 
-        if (pending_matches[pending_matches.length - 1].game_count >= match.number_to_play) pending_matches.pop();
+        if (match.game_count >= match.number_to_play) pending_matches.pop();
```

Comment:
It was a hack basically that I'd add requests to game_count and then pop if game_count + requests was more than we needed. But then we can end up with some missing, etc. If we instead don't pop until game_count really did get those games in, it'll be more accurate and next match won't start. It'd slow down next match starting, but then we won't queue a new match against wrong best_network if this network promotes on like game 399 or 400.

Reply:
Yeah, the next match shouldn't start before it should start anyway. So that is ok. If you want faster matches you can bump QUEUE_BUFFER. However, i expect matches will already go a lot faster after this patch, and you overshoot a bit more, so in practice you might want to decrease QUEUE_BUFFER to get less overshoot. Another value you can later play with is MATCH_EXPIRE_TIME.